gpio/zynq7000-pwm/pwm.c

```diff
@@ -138,6 +143,29 @@ static void pwm_thread(void *arg)
 				msg.o.attr.err = EOK;
 				break;
 
+			case mtDevCtl:
+				if (iptr->type == zynq7000pwm_msgGet) {
+					for (i = 0; i < sizeof(iptr->compval); ++i) {
```

Comment:
Index exceeds the number of elements in both arrays `pwm_common.channel[8]` and `optr->compval[8]`.
I believe you meant  either `sizeof(iptr->compval) / sizeof(*iptr->compval) = 8` or `sizeof(iptr->compval) / sizeof(iptr->compval[0]) = 8` instead of `sizeof(iptr->compval) = 8 * 4`.
btw. also `optr` instead of `iptr`.
```suggestion
					for (i = 0; i < sizeof(optr->compval) / sizeof(*optr->compval); ++i) {
```
or use `PWM_NCHAN`
```suggestion
					for (i = 0; i < PWM_NCHAN; ++i) {
```

Reply:
Fixed in https://github.com/phoenix-rtos/phoenix-rtos-devices/pull/340, thanks!